ui/utils/format.ts

```diff
@@ -1,17 +1,28 @@
-const K = 1024;
-const UNITS = ['B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];;
+import convert from "convert";
+import { useFormatter } from "next-intl";
 
-export function formatBytes(bytes: number | null, decimals: number) {
-    if (bytes == null) {
-        return '-';
-    }
-
-    if (bytes == 0) {
-        return '0 B';
-    }
-
-    let dm = decimals || 2;
-    let i = Math.floor(Math.log(bytes) / Math.log(K));
-
-    return parseFloat((bytes / Math.pow(K, i)).toFixed(dm)) + ' ' + UNITS[i];
+export function useFormatBytes() {
+  const format = useFormatter();
+  return function formatBytes(bytes: number) {
+    const res = convert(bytes, "bytes").to("best");
+    const unit: Record<(typeof res)["unit"], string> = {
+      bits: "bit",
+      GiB: "gigabyte",
+      KiB: "kilobyte",
+      GB: "gigabyte",
+      MiB: "megabyte",
+      PiB: "petabyte",
+      TiB: "terabyte",
```

Comment:
Should these be `"gibibyte"`, `"kibibyte"`, etc.?

Reply:
yes! And no. Or maybe? The intl api doesn't support the iB units, so we have to use the bytes ones. Which is ok, as we have the data in bytes, thus we'll not get the iB variants and those mappings will never be used.

Although one could argue that we should display the data in KiB et friends instead of KB; checking the Wikipedia KB is base 10, where KiB are base 2.

I hate this stuff 🙃

WDYT, should we use KiB or KB?

Reply:
I guess it depends on what users mostly expect. I think the ones that use the traditional prefixes kilo, mega, etc. are sometimes base 10 and sometimes base 2 whereas the iB units are explicitly base 2. This may be something to raise for discussion as a general topic so that it aligns completely within the console and with other components that may be involved in a streaming platform.

Reply:
I see now on the Wikipedia page that the traditional SI units should be limited base 10, but I guess that doesn't prevent it from being ambiguous to individuals still :-) Ultimately, I don't mind whatever way we go as long as it's consistent.

Reply:
Maybe for now we can just remove the iB variants if they'll never be used by this logic as it is. wdyt?

Reply:
Ok, we should definitely come back and change that based on the outcome of the ADR.

Reply:
I changed the code so it's clear what we are doing. We are getting bytes in input, we are outputting them in *iB (so, base 2).

Eg. 10875778 bytes is 10.372 MiB

Reply:
LGTM - so "imperial" means the *iB units? btw, I'm looking at the test that is _still_ flaky somehow.

Reply:
Yeah, TIL. But it's specific to this library, probably it's like so because it uses a common interface for all units that it supports and it started supporting things like distances and weights? Not sure 🤷